### PR TITLE
Make expansion name testing against the binding name case insensitive.

### LINKED ISSUE #68: Query options are ignored in Expand except $select
Thanks for the great work with this great library.

I have already created a [StackOverflow question](https://stackoverflow.com/questions/64568079/invalidoperationexception-with-top-or-skip-in-odata-expand) related with this issue and wanted to also add it here in case it is missed there.

Query options inside the Expand is completely ignored while it partially works (except for $skip and $top) with ProjectTo.

https://localhost:44347/v1/odata/subcategories?$top=2&$expand=products($top=1) is ignored and all nested products are expanded

Here is how we use it with the Queryable Extensions and all the 
```
        [AllowAnonymous, HttpGet]
        public async Task<IActionResult> Get(ODataQueryOptions<ProductSubcategoryDto> options)
        {
             return Ok(await DbContext.ProductSubcategories.GetQueryAsync(Mapper, options));
        }
```

The only option that seems to not be ignored is $select.

Am I doing everything right here? Is there a reason for these query options to be ignored?

We are using EF Core 3.1.9 with Automapper 10.1.1 and AutoMapper.AspNetCore.OData.EFCore 2.0.2.

### PR DESCRIPTION
**Pull request by hasandogu**
Fix #68

### CODE REVIEW

**Comment by hasandogu** [comment]
Hey @BlaiseD 

What would I need to do to get this through? It is the same failure as the other PR #59:

-Missing value for option 'api-key'

**Comment by BlaiseD** [comment]
Fixed the build error to "not push" on fork.

File: AutoMapper.OData.EFCore.Tests/GetQueryTests.cs [hunk 1]
```diff
@@ -556,7 +556,7 @@ public async void FilteringOnRoot_ChildCollection_AndChildCollectionOfChildColle
             (
                 await Get<CategoryModel, Category>
                 (
-                    "/CategoryModel?$top=5&$expand=Products($filter=SupplierAddress/City ne '';$orderby=ProductName;$expand=AlternateAddresses($filter=City ne '';$orderby=City desc),SupplierAddress)&$filter=CategoryName ne ''&$orderby=CategoryName asc",
+                    "/CategoryModel?$top=5&$expand=Products($filter=SupplierAddress/City ne '';$orderby=ProductName;$expand=AlternateAddresses($filter=City ne '';$orderby=City desc;$top=1),SupplierAddress)&$filter=CategoryName ne ''&$orderby=CategoryName asc",
```
Comment (BlaiseD) [inline AutoMapper.OData.EFCore.Tests/GetQueryTests.cs:559]: This change works with or without the fix in ProjectionVisitor.  Maybe add a separate test with a case-insensitive URL?

**Review by BlaiseD — CHANGES_REQUESTED** [review]
The new test should fail without the change and work because of the change.

File: AutoMapper.OData.EFCore.Tests/GetQueryTests.cs (hunk 1 above)
Comment (hasandogu) [reply AutoMapper.OData.EFCore.Tests/GetQueryTests.cs:559]: Actually, the URL casing doesn't trigger this problem as the parser seem to normalize them to the property names in the EDM so they work in any casing.

Enabling lower camel case changes the name of the properties in the EDM, which then triggers this error. I thought we could simply add this to the query tests as I had not found a test case where $top or $skip was called, which seem to now be added.

I could certainly add a new test (to the web tests only) but need to enable the lower camel case by default. Would that be ok?

Comment (BlaiseD) [reply AutoMapper.OData.EFCore.Tests/GetQueryTests.cs]: Not sure what you mean by enabling by default, but you're saying that the configuration and not the case of the URL itself causes the error, then multiple existing test will fail if we enable  lower camel case. The PR should be just two lines of code in that case.

- The fix.
- Add `builder.EnableLowerCamelCase();` [here](https://github.com/hasandogu/AutoMapper.Extensions.OData/blob/a8f6795ba358bec334f0c7b186bd6b6aed47eb1c/AutoMapper.OData.EFCore.Tests/GetTests.cs#L436) (so tests fail without the fix).

Comment (BlaiseD) [reply AutoMapper.OData.EFCore.Tests/GetQueryTests.cs]: We should already have a test with the $take for child collections.   Your fork is a few commits behind.

**Review by BlaiseD — APPROVED** [review]

**Comment by hasandogu** [comment]
Thanks @BlaiseD. Yes, the tests would fail without the fix. We should also consider Enabling the lower camel case for the web tests by enabling them here as the tests would fail without the fix.

https://github.com/hasandogu/AutoMapper.Extensions.OData/blob/a8f6795ba358bec334f0c7b186bd6b6aed47eb1c/WebAPI.OData.EFCore/Startup.cs#L71

and here

https://github.com/hasandogu/AutoMapper.Extensions.OData/blob/a8f6795ba358bec334f0c7b186bd6b6aed47eb1c/WebAPI.AspNet.OData.EF6/App_Start/WebApiConfig.cs#L33

Want me to add to this PR?

**Comment by BlaiseD** [comment]
No worries.  Sample projects reference the packages i.e. you'd need the new package anyway for the tests to pass.

**Comment by hasandogu** [comment]
Any change to push a preview release or should we rely on MyGet?

**Comment by BlaiseD** [comment]
[2.0.3-preview.0](https://www.nuget.org/packages/AutoMapper.AspNetCore.OData.EFCore/2.0.3-preview.0).

**Comment by hasandogu** [comment]
Thanks @BlaiseD

**Pull request merged**